src/licensedcode/match.py

```diff
@@ -1485,6 +1540,10 @@ def _log(_matches, _discarded, msg):
         all_discarded.extend(discarded)
         _log(matches, discarded, 'HIGH ENOUGH SCORE')
 
+    matches, discarded = filter_key_phrase_spans(matches)
```

Comment:
I think this filter should be positioned much earlier in the process otherwise here is the risk:
1. there was a smalller match that was contained
2. it is filtered because of containment in a larger match
3. the larger is discarded because of not having keyphrases
4. there is no match left for this region and we now have a false negative

Reply:
I've moved the key phrase filter up in 2c377b3a6bd69920a7dfb68864d3e7f9bf58901b, to right before the first containment filter. I was unable to reproduce this scenario exactly, but some datadriven tests at least demonstrated the effectiveness (`tests/licensedcode/test_detection_datadriven1.py::TestLicenseDataDriven1::test_detection_datadriven_lic1_alexa_skills_kit_sdk_for_java_txt`)

Reply:
@pombredanne Would the above suffice?